plugins/state_monitor.py

```diff
@@ -0,0 +1,238 @@
+            #     self.logger.info(msg)
+            #     self.send_message(msg)
+
+            # LED state
+            # todo detect this more reliably by checking array_fault and load_fault bitfields are non-zero,
+            state_now, state_pre = now_then('led_state')
+            if (state_now >= 12) and (state_now != state_pre):
+                msg = f"Critical failure with charge controller. LED state: {state_now}."
+                self.logger.info(msg)
+                self.send_message(msg)
+
+        # If expression evaluation fails, presumably due to bad data values
+        except Exception as e:
+            self.logger.error(
+                "%s evaluating in %s on step %s: %s",
+                type(e).__name__, type(self), self.name, e)
+            self.logger.info("Error details:", exc_info=True)
+            for pretty_name, data in [("Current", input_data),
+                                      ("Previous", self._previous_data)]:
+                self.logger.info(
+                    "%s data: %r", pretty_name,
+                    {key: str(value) for key, value in data.items()})
+
+        # Update state for next pass through the pipeline
+        self._previous_data = input_data
+
+        # Pass through the input for consumption by any further steps
+        return input_data
+
+    def send_message(self, msg):
+        """
+        Send a message via the given method.
+
+        This is a shepherd method. Given a generic message, we'll send it
+        using the method specified when initializing the class. Before sending, we'll
+        add in which sensor is sending the message.
+
+        Note: Any "sender class" must have a send method.
+
+        Parameters
+        ----------
+        msg : str
+            The message to be sent.
+
+        """
+        msg = self.construct_message(msg)
+
+        if self.method_cls is not None:
+            self.method_cls.send(msg)
+
+    @classmethod
+    def construct_message(cls, msg):
+        """ Construct the message detailing the state notification. """
+        from brokkr.config.unit import UNIT_CONFIG
+        from brokkr.config.metadata import METADATA
+
+        sensor_name = f"{METADATA['name']}{UNIT_CONFIG['number']:02d}"
+
+        header = f"Message from sensor {sensor_name}"
+
+        msg = '\n'.join([header, msg])
+
+        return msg
+
+
+class SlackSender:
+    def __init__(self, key_file, channel, logger=None):
+        """ 
+        Handle sending messages in Slack. 
+        
+        Parameters
+        ----------
+        key_file : str
+            The path to the file that contains the Slack key for the
+            workspace to send the message.
+        channel : str
+            The channel to send the message to. It should be defined in file
+            provided by `key_file`.
+        logger : Python logger
+            If not None, if message send fails, log it here.
```

Comment:
```suggestion
        logger : logging.Logger or False, optional
            If not False and message send fails, log it to the passed logger
            or the default logger, if not passed.
```

Revise this to use the default logger by default.